libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -501,6 +512,65 @@ void AP_Camera::take_picture()
     GCS_MAVLINK::send_to_components(MAVLINK_MSG_ID_COMMAND_LONG, (char*)&cmd_msg, sizeof(cmd_msg));
 }
 
+// start/stop recording video
+// start_recording should be true to start recording, false to stop recording
+void AP_Camera::record_video(bool start_recording)
+{
+#if HAL_MOUNT_ENABLED
+    // only mount implements recording video
+    if (get_trigger_type() == CamTrigType::mount) {
+        AP_Mount* mount = AP::mount();
+        if (mount != nullptr) {
+            mount->record_video(0, start_recording);
+        }
+    }
+#endif
+}
+
+// zoom in, out or hold
+// zoom out = -1, hold = 0, zoom in = 1
+void AP_Camera::set_zoom_step(int8_t zoom_step)
+{
+#if HAL_MOUNT_ENABLED
+    // only mount implements set_zoom_step
+    if (get_trigger_type() == CamTrigType::mount) {
+        AP_Mount* mount = AP::mount();
+        if (mount != nullptr) {
+            mount->set_zoom_step(0, zoom_step);
+        }
+    }
+#endif
+}
+
+// focus in, out or hold
+// focus in = -1, focus hold = 0, focus out = 1
+void AP_Camera::set_manual_focus_step(int8_t focus_step)
+{
+#if HAL_MOUNT_ENABLED
+    // only mount implements set_manual_focus_step
+    if (get_trigger_type() == CamTrigType::mount) {
+        AP_Mount* mount = AP::mount();
+        if (mount != nullptr) {
+            mount->set_manual_focus_step(0, focus_step);
+        }
+    }
+#endif
+}
+
+// auto focus
+void AP_Camera::set_auto_focus()
+{
+#if HAL_MOUNT_ENABLED
+    // only mount implements set_auto_focus
+    if (get_trigger_type() == CamTrigType::mount) {
+        AP_Mount* mount = AP::mount();
```

Comment:
```suggestion
        AP_Mount_Siyi* mount = AP::mount_siyi();
```

.... and create the singleton-getter in the backend.  This is probably how we should have done the Solo camera/gimbal combination.  This will allow removal of the virtual methods on Mount and mount-backend.

I actually had a similar problem with the Loweheiser generator, where it acts as both an EFI and Generator backend.  I will probably change it to use this same pattern (assuming people are good with it), rather than the rather ugly method I'm currently using in that PR.

Reply:
I initially implemented it as you've suggested using a direct singleton to the siyi driver but I think longer-term it is better to access these commands via the gimbal.  I think there are likely to be other gimbals which include cameras and going via the AP_Mount library allows us to keep the instances consistent.  E.g. camera instance 1 will be tied to mount instance 1... of course AP_Camera doesn't have instances yet but I hope to add this in a future PR.

Reply:
STorM32 gimbal controller is also a camera controller since its existance and thus would be another candidate. It IMHO should clearly go through the mount frontend. my 2 cents